gsa/src/web/pages/credentials/component.js

```diff
@@ -26,192 +26,196 @@ import {shorten} from 'gmp/utils/string';
 import {loadUserSettingDefaults} from 'web/store/usersettings/defaults/actions';
 import {getUserSettingsDefaults} from 'web/store/usersettings/defaults/selectors';
-import {getUsername} from 'web/store/usersettings/selectors';
-
-import compose from 'web/utils/compose';
 
+import PropTypes from 'web/utils/proptypes';
 import {generateFilename} from 'web/utils/render';
+import useGmp from 'web/utils/useGmp';
+import useUserName from 'web/utils/useUserName';
+import useUserSessionTimeout from 'web/utils/useUserSessionTimeout';
 
 import CredentialsDialog from './dialog';
 
-class CredentialsComponent extends React.Component {
-  constructor(...args) {
-    super(...args);
-
-    this.state = {dialogVisible: false};
-
-    this.handleCloseCredentialDialog = this.handleCloseCredentialDialog.bind(
-      this,
-    );
-    this.openCredentialsDialog = this.openCredentialsDialog.bind(this);
-    this.handleDownloadInstaller = this.handleDownloadInstaller.bind(this);
-  }
-
-  openCredentialsDialog(credential) {
-    if (isDefined(credential)) {
-      const title = _('Edit Credential {{name}}', {
-        name: shorten(credential.name),
-      });
-
-      this.setState({
-        allow_insecure: credential.allow_insecure,
-        comment: credential.comment,
-        credential,
-        credential_type: credential.credential_type,
-        auth_algorithm: credential.auth_algorithm,
-        name: credential.name,
-        credential_login: credential.login,
-        privacy_algorithm: isDefined(credential.privacy)
-          ? credential.privacy.algorithm
-          : undefined,
-        types: [credential.credential_type],
-        dialogVisible: true,
-        title,
-      });
-    } else {
-      // reset all values in state to not show values from last edit
-      this.setState({
-        allow_insecure: undefined,
-        comment: undefined,
-        credential: undefined,
-        credential_type: undefined,
-        auth_algorithm: undefined,
-        name: undefined,
-        credential_login: undefined,
-        privacy_algorithm: undefined,
-        types: ALL_CREDENTIAL_TYPES,
-        dialogVisible: true,
-        title: _('New Credential'),
-      });
-    }
-
-    this.handleInteraction();
-  }
-
-  closeCredentialDialog() {
-    this.setState({dialogVisible: false});
-  }
-
-  handleCloseCredentialDialog() {
-    this.closeCredentialDialog();
-    this.handleInteraction();
-  }
-
-  handleDownloadInstaller(credential, format) {
-    const {
+const CredentialsComponent = ({
+  children,
+  onCreateError,
+  onCreated,
+  onCloneError,
+  onCloned,
+  onDeleteError,
+  onDeleted,
+  onDownloadError,
+  onDownloaded,
+  onInstallerDownloadError,
+  onInstallerDownloaded,
+  onSaved,
+  onSaveError,
+}) => {
+  const dispatch = useDispatch();
+  const userDefaultsSelector = useSelector(getUserSettingsDefaults);
+  const userName = useUserName();
+  const gmp = useGmp();
+  const [, renewSessionTimeout] = useUserSessionTimeout();
```

Comment:
In our previous PR I think we called this function `renewSession`, though I don't have particularly strong feelings on this matter.

Reply:
It's just a variable name here. We could have called it foo here too ;-)